src/Ledger/Ledger/Properties.agda

```diff
@@ -82,29 +102,579 @@ instance
+      fromList (dpMap (propUpdate (propUpdate govSt y k) x (suc k) ++ (updateGovStates (map inj₂ ys) (suc (suc k)) [])))
+        ≈˘⟨ ≡ᵉ.reflexive (cong ((fromList ∘ dpMap)) (updateProps-decomp ys)) ⟩
+      fromList (dpMap (updateGovStates (map inj₂ ys) _ (propUpdate (propUpdate govSt y k) x (suc k))))
+        ∎
+
+
+    -- GA Deposits Invariance Property for LEDGER STS --------------------------------------------------------------------
+    LEDGER-govDepsMatch : ∀ {s' : LState} → Γ ⊢ s ⇀⦇ tx ,LEDGER⦈ s'
+                          → govDepsMatch s → govDepsMatch s'
+    LEDGER-govDepsMatch (LEDGER-I⋯ refl (UTXOW-UTXOS (Scripts-No _))) aprioriMatch = aprioriMatch
+
+    LEDGER-govDepsMatch s'@{⟦ .(⟦ ((UTxOState.utxo (LState.utxoSt s) ∣ txins ᶜ) ∪ˡ (outs txb))
+                                , _ , updateDeposits pp txb (UTxOState.deposits (LState.utxoSt s)) , _ ⟧ᵘ)
+                            , govSt' , _ ⟧ˡ}
+      utxosts@(LEDGER-V⋯ tx-valid (UTXOW-UTXOS (Scripts-Yes x)) _ GOV-sts) aprioriMatch = begin
+        filterˢ isGADeposit (dom (updateDeposits pp txb utxoDeps))
+          ≈⟨ noGACerts txcerts (updateProposalDeposits txprop txid govActionDeposit utxoDeps) ⟩
+        filterˢ isGADeposit (dom (updateProposalDeposits txprop txid govActionDeposit utxoDeps))
+          ≈⟨ filter-pres-≡ᵉ (updatePropDeps≡ᵉ txprop) ⟩
+        filterˢ isGADeposit (dom (utxoDeps ∪⁺ proposalDepositsΔ txprop pp txb))
+          ≈⟨ filter-pres-≡ᵉ dom∪⁺ ⟩
+        filterˢ isGADeposit ((dom utxoDeps) ∪ dom (proposalDepositsΔ txprop pp txb))
+          ≈⟨ filter-hom-∪ ⟩
+        filterˢ isGADeposit (dom utxoDeps) ∪ filterˢ isGADeposit (dom (proposalDepositsΔ txprop pp txb))
+          ≈⟨ ∪-cong aprioriMatch (allGA-propDepsΔ txprop) ⟩
+        fromList (dpMap govSt) ∪ dom (proposalDepositsΔ txprop pp txb ˢ)
+          ≈⟨ ∪-cong ≡ᵉ.refl (utxo-govst-connex txprop) ⟩
+        fromList (dpMap govSt) ∪ fromList (dpMap (updateGovStates (map inj₂ txprop) 0 []))
+         ≈˘⟨ ∪-cong ≡ᵉ.refl (≡ᵉ.reflexive (cong fromList (dmMap-vote-invar++ txprop txvote {0}))) ⟩
+        fromList (dpMap govSt) ∪ fromList (dpMap (updateGovStates (map inj₂ txprop ++ map inj₁ txvote) 0 []))
+          ≈⟨ ∪-fromList-++ (dpMap govSt) (dpMap (updateGovStates (map inj₂ txprop ++ map inj₁ txvote) 0 []))⟩
+        fromList (dpMap govSt ++ dpMap (updateGovStates (map inj₂ txprop ++ map inj₁ txvote) 0 []))
+          ≈˘⟨ ≡ᵉ.reflexive (cong fromList (map-++ (λ (id , _) → GovActionDeposit id) govSt (updateGovStates (txgov txb) 0 []))) ⟩
+        fromList (dpMap (govSt ++ updateGovStates (txgov txb) 0 []))
+          ≈˘⟨ ≡ᵉ.reflexive (cong fromList (updateGovStates≡ (txgov txb) 0)) ⟩
+        fromList (dpMap (updateGovStates (txgov txb) 0 govSt ))
+          ≈˘⟨ ≡ᵉ.reflexive (cong (fromList ∘ dpMap ) (STS→GovSt≡ utxosts tx-valid)) ⟩
+        fromList (dpMap govSt')
+          ∎
+        where
+        utxo-govst-connex : (props : List GovProposal)
```

Comment:
Yeah, it's a bit difficult. It's more about how specific and big something is than whether it's used by just that thing. So if it's just used in one place, there's a higher probability that it's also specific to that place, but it might not be. And this lemma looks like it might be useful to prove other things.